stores/metadata.go

```diff
@@ -664,24 +664,25 @@ func (s *SQLStore) ArchiveContract(ctx context.Context, id types.FileContractID,
 }
 
 func (s *SQLStore) ArchiveContracts(ctx context.Context, toArchive map[types.FileContractID]string) error {
-	// fetch ids
-	var ids []types.FileContractID
-	for id := range toArchive {
-		ids = append(ids, id)
-	}
+	var errs []string
+	for fcid, reason := range toArchive {
+		// invalidate health of related sectors before archiving the contracts
+		// NOTE: even if this is not done in the same transaction it won't have any
+		// lasting negative effects.
+		if err := s.invalidateSlabHealthByFCID(ctx, []types.FileContractID{fcid}); err != nil {
```

Comment:
The main thing is archiving one by one right, can't we still get away with invalidating the successfully archived contracts in batch?

Reply:
The invalidations are faster. Almost negligible by comparison. So I'm performing them right before archiving a contract to make sure that not a lot of time passes in between. Considering that we don't perform the invalidation in the same txn as the archival.